template/src/pageComponents/buy-me-coffee/steps/BuyCoffeeFormStep/BuyCoffeeFormStep.tsx

```diff
@@ -1,30 +1,42 @@
+import { RefetchOptions, QueryObserverResult } from '@tanstack/react-query';
+import { ReadContractErrorType } from 'viem';
 import { TransactionSteps } from '../../ContractDemo';
 import FormBuyCoffee from './FormBuyCoffee';
 
 type BuyCoffeeFormStepProps = {
-  onComplete: () => void;
   setTransactionStep: React.Dispatch<React.SetStateAction<TransactionSteps | null>>;
   numCoffees: number;
+  transactionStep: TransactionSteps | null;
   setNumCoffees: React.Dispatch<React.SetStateAction<number>>;
+  refetchMemos: (options?: RefetchOptions | undefined) => Promise<
+    QueryObserverResult<
+      readonly {
```

Comment:
feel free to ignore, i'm being pedantic. But you can also use `abitype` to get input arg types from abi like so instead of hardcoding them

```
const RefetchMemoInputArgs = ExtractAbiFunction<typeof BuyMeACoffeeABI, 'refetchMemos'>['inputs']
```

Reply:
Ahh nice! Okay will update this as well soon!

Reply:
Actually correct way is 

```
AbiParametersToPrimitiveTypes<
        ExtractAbiFunction<typeof BuyMeACoffeeABI, 'getMemos'>['outputs']
      >[0],
```